src/app/components/abstract/abstract-form-parent-base/readme.md

```diff
@@ -0,0 +1,100 @@
+
+## Table of Contents
+
+- [Installation](#installation)
+- [Usage](#usage)
+- [Inputs](#inputs)
+- [Outputs](#outputs)
+- [Methods](#methods)
+- [Testing](#testing)
+- [Contributing](#contributing)
+
+## Installation
+
+To use the `AbstractFormParentBaseComponent` in your project, ensure that it is extended by parent components that manage child form components or controls.
+
+```typescript
+import { AbstractFormParentBaseComponent } from '@components/abstract/abstract-form-parent-base/abstract-form-parent-base.component';
+```
+
+## Usage
+
+This component is designed to be used as a base class for components that need to manage forms with child form controls or components. It facilitates communication between parent and child forms for validation, form submission, and error handling.
+
+### Example Usage:
+
+```typescript
+import { Component } from '@angular/core';
+import { AbstractFormParentBaseComponent } from '@components/abstract/abstract-form-parent-base/abstract-form-parent-base.component';
+
+@Component({
+  selector: 'app-my-parent-form',
+  templateUrl: './my-parent-form.component.html',
+})
+export class MyParentFormComponent extends AbstractFormParentBaseComponent {
+  // Define child form logic and interaction methods
+}
+```
+
+## Inputs
+
+The `AbstractFormParentBaseComponent` facilitates managing multiple child form components or form controls. Components that extend this class typically manage form groups and handle parent-child form interactions.
+
+| Input         | Type       | Description                                                 |
+| ------------- | ---------- | ----------------------------------------------------------- |
+| `formGroup`   | `FormGroup`| The form group that manages the parent form and child controls.|
+
+## Outputs
+
+The component may provide outputs to notify the parent component of changes in child forms or the form state. For instance, it can notify when a child form is submitted or validated.
+
+## Methods
+
+`AbstractFormParentBaseComponent` provides utility methods to manage the interaction between parent and child forms, handle validation, and manage form submissions.
+
+### Common Methods:
+
+- **`initializeChildForm()`**: Initializes child form components or controls within the parent form.
+- **`validateChildForms()`**: Runs validation on all child forms or form controls.
+- **`getChildFormErrors()`**: Retrieves error messages from child forms or controls.
+- **`submitParentForm()`**: Handles form submission for the parent form and triggers child form validation.
+
+Example:
+
+```typescript
+this.initializeChildForm();
+const childErrors = this.getChildFormErrors();
+if (this.validateChildForms()) {
+  this.submitParentForm(); // Proceed with form submission
+}
+```
```

Comment:
Make sure this section is relevant to the abstract class as some of these methods do not exist in the abstract class